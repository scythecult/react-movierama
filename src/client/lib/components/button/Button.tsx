import clsx from 'clsx';
import type { ButtonHTMLAttributes, PropsWithChildren } from 'react';
import type { PropsWithClassName } from '../../types/PropsWithClassName';
import styles from './styles.module.css';

export type ButtonProps = PropsWithClassName & PropsWithChildren<ButtonHTMLAttributes<HTMLButtonElement>>;
export const Button = (props: ButtonProps) => {
  const { className, children, ...restProps } = props;
  const classNameFinal = clsx(className, styles.button);

  return (
    <button className={classNameFinal} {...restProps}>
      {children}
    </button>
  );
};
