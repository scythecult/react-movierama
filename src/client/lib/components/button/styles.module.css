.button {
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 8px 16px;
    color: var(--app-white-color);
    background-color: var(--app-primary-button);
    border-radius: var(--app-border-radius-s);
    border-color: transparent;
    transition-property: opacity, background-color;
    transition-duration: 0.2s;
}

.button:hover:not(:disabled) {
    background-color: var(--app-primary-button-hover);
    cursor: pointer;
}

.button:disabled {
    opacity: 0.5;
    cursor: not-allowed;
}