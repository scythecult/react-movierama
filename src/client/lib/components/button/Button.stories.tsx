import type { Meta, StoryFn } from '@storybook/react-vite';
import { Button, type ButtonProps } from './Button';

export default {
  title: 'Components/Button',
  component: Button,
  args: {
    children: 'Check',
  },
} satisfies Meta<typeof Button>;

const Template: StoryFn<ButtonProps> = (props) => <Button {...props} />;

export const Default = Template.bind(null);
Default.args = {
  disabled: true,
};
