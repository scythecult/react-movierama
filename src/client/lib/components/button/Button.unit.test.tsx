import { render } from '@testing-library/react';
import { Button } from '@/client/lib/components/button/Button';

describe('Button', () => {
  test('should correspond default layout', () => {
    const result = render(<Button />);

    expect(result.container).toMatchSnapshot();
  });

  test('should support the "children" prop', () => {
    const result = render(<Button>Children</Button>);

    expect(result.container).toMatchSnapshot();
  });

  test('should support the "className" prop', () => {
    const result = render(<Button className="custom-class">Children</Button>);

    expect(result.container).toMatchSnapshot();
  });
});
