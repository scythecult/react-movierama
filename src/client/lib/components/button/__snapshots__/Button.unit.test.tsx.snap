// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should correspond default layout 1`] = `
<div>
  <button
    class="_button_d1d523"
  />
</div>
`;

exports[`Button > should support the "children" prop 1`] = `
<div>
  <button
    class="_button_d1d523"
  >
    Children
  </button>
</div>
`;

exports[`Button > should support the "className" prop 1`] = `
<div>
  <button
    class="custom-class _button_d1d523"
  >
    Children
  </button>
</div>
`;
